File: Shared/Dto/Underlying/ForWebApi/RtuDto.cs
using System;

namespace Iit.Fibertest.Dto
{
    public class RtuDto
    {
        public Guid RtuId;
        public RtuMaker RtuMaker;
        public string Title;

        public string Mfid;
        public string Mfsn;
        public string Omid;
        public string Omsn; 
        public string Serial;

        public int OwnPortCount;
        public int FullPortCount;
        public OtauDto[] Otaus;
        public TraceDto[] Traces;

        public NetAddress MainChannel;
        public RtuPartState MainChannelState;
        public NetAddress ReserveChannel;
        public RtuPartState ReserveChannelState;
        public bool IsReserveChannelSet;
        public NetAddress OtdrNetAddress;
        public RtuPartState BopState;

        public MonitoringState MonitoringMode;

        public string Version;
        public string Version2;

    }
}

File: AngularClientApp/src/app/models/dtos/rtuDto.ts
import { MonitoringMode } from '../enums/monitoringMode';
import { RtuMaker } from '../enums/rtuMaker';
import { NetAddress } from './netAddress';
import { RtuPartState } from '../enums/rtuPartState';
import { ChildDto } from './childDto';

export interface RtuDto {
  rtuId: string;
  rtuMaker: RtuMaker;
  title: string;

  mfid: string;
  mfsn: string;
  omid: string;
  omsn: string;

  fullPortCount: number;
  ownPortCount: number;

  children: ChildDto[];

  mainChannel: NetAddress;
  mainChannelState: RtuPartState;
  reserveChannel: NetAddress;
  reserveChannelState: RtuPartState;
  isReserveChannelSet: boolean;
  otdrNetAddress: NetAddress;
  bopState: RtuPartState;

  monitoringMode: MonitoringMode;

  version: string;
  version2: string;
}
